File: 백준/Silver/1850. 최대공약수/최대공약수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    private static long gcd(long a, long b) {
        return b == 0 ? a : gcd(b, a % b);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        long a = Long.parseLong(st.nextToken());
        long b = Long.parseLong(st.nextToken());

        long gcd = gcd(a, b);

        StringBuilder sb = new StringBuilder();
        for (long i = 0; i < gcd; i++) {
            sb.append('1');
        }

        System.out.print(sb);
    }
}

File: 백준/Silver/1850. 최대공약수/최대공약수.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.StringTokenizer

private fun gcd(a: Long, b: Long): Long = if (b == 0L) a else gcd(b, a % b)

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val st = StringTokenizer(br.readLine())
    val a = st.nextToken().toLong()
    val b = st.nextToken().toLong()

    val gcd = gcd(a, b)
    println("1".repeat(gcd.toInt()))
}
